library/common.ts

```diff
@@ -50,3 +51,20 @@ export const IncomingInvoiceTableMap: Record<
   [IncomingInvoiceType.LNURL_Zap]: "transaction",
   [IncomingInvoiceType.LNURL]: "transaction",
 };
+
+export interface PromoResponseData {
+  user: PromoResponseUser;
+  id: number;
+  msatBudget: number;
+  msatPayoutAmount: number;
+  contentId: string;
+  contentType: string;
+  contentMetadata?: TrackInfo;
+}
+
+export interface PromoResponseUser {
+  canEarnToday: boolean;
+  lifetimeEarnings: number;
+  earnedToday: number;
+  earnableToday: number;
+}
```

Comment:
looks like the controllers all use naming that are slightly different
```suggestion
export interface PromoResponseData {
  promoUser: PromoResponseUser;
  id: number;
  msatBudget: number;
  msatPayoutAmount: number;
  contentId: string;
  contentType: string;
  contentMetadata?: TrackInfo;
}

export interface PromoResponseUser {
  canEarnToday: boolean;
  lifetimeEarnings: number;
  totalEarnedToday: number;
  availableEarnings: number;
}
```